### PR TITLE
KTOR-7284 Deprecate withTestApplication and withApplication

### PR DESCRIPTION
**Pull request by osipxd**
**Subsystem**
Testing Infrastructure

**Motivation**
[KTOR-7284](https://youtrack.jetbrains.com/issue/KTOR-7284/Deprecate-withTestApplication-APIs-with-ERROR-and-migrate-to-testApplication) Deprecate `withTestApplication` APIs with ERROR

**Solution**
Removed the rest of `withApplication` and `withTestApplication` usages and changed deprecation level to `ERROR`.

### CODE REVIEW

**Review by marychatte — APPROVED** [review]
Good 👍

File: ktor-server/ktor-server-tests/jvm/test/io/ktor/server/testing/TestApplicationEngineTest.kt [hunk 1]
```diff
@@ -41,247 +41,245 @@ class TestApplicationEngineTest {
...
+        }.response
+        assertEquals(HttpStatusCode.OK, response.status())
+    }
+
+    private suspend fun ApplicationTestBuilder.startApplicationAndGetEngine(): TestApplicationEngine {
+        var engine: TestApplicationEngine? = null
+        application { engine = this.engine as TestApplicationEngine }
+        startApplication()
+        return checkNotNull(engine)
     }
```
[the remark this answers is not in the document]
Comment (bjhham) [reply ktor-server/ktor-server-tests/jvm/test/io/ktor/server/testing/TestApplicationEngineTest.kt:283]: Yeah, from what I can tell it ought to be marked as `@Internal` or possibly `internal` with the keyword.  The idea here is that you shouldn't have to interact with it at all, and be able to do everything through TestApplication.

**Review by bjhham — APPROVED** [review]

**Comment by osipxd** [comment]
I hope I didn't delete anything important 🤞

**Review by bjhham — APPROVED** [review]
Red stuff in the API looks good to me 👍

**Pull request merged**